File: core-sample-app/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/sampleapp/examples/chromecastExample/CastOptionsProvider.java
package com.pierfrancescosoffritti.androidyoutubeplayer.core.sampleapp.examples.chromecastExample;

import android.content.Context;

import com.google.android.gms.cast.framework.CastOptions;
import com.google.android.gms.cast.framework.OptionsProvider;
import com.google.android.gms.cast.framework.SessionProvider;

import java.util.List;

/**
 * Class providing setup info to the Chromecast framework, declared in manifest file.
 *
 * <a href="https://developers.google.com/cast/docs/android_sender_integrate#initialize_the_cast_context">See doc here</a>
 */
public final class CastOptionsProvider implements OptionsProvider {
    public CastOptions getCastOptions(Context appContext) {
        /*
          This is the receiver id of the sample receiver.
          Remember to change it with the ID of your own receiver. See documentation for more info.
          <a href="https://github.com/PierfrancescoSoffritti/chromecast-youtube-player">chromecast-youtube-player</a>
          */
        String receiverId = "C5CBE8CA";

        return new CastOptions.Builder()
                .setReceiverApplicationId(receiverId)
                .build();
    }

    public List<SessionProvider> getAdditionalSessionProviders(Context context) {
        return null;
    }
}

File: chromecast-sender-sample-app/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/chromecast/sampleapp/examples/CastOptionsProvider.kt
package com.pierfrancescosoffritti.androidyoutubeplayer.chromecast.sampleapp.examples

import android.content.Context
import com.google.android.gms.cast.framework.CastOptions
import com.google.android.gms.cast.framework.SessionProvider
import com.google.android.gms.cast.framework.OptionsProvider

/**
 * Class providing setup info to the Chromecast framework, declared in manifest file.
 *
 * [see doc here](https://developers.google.com/cast/docs/android_sender_integrate#initialize_the_cast_context)
 */
internal class CastOptionsProvider : OptionsProvider {
    // This is the receiver id of the sample receiver.
    // Remember to change it with the ID of your own receiver. See documentation for more info.
    private val receiverId = "C5CBE8CA"

    override fun getCastOptions(appContext: Context): CastOptions {
        return CastOptions.Builder()
                .setReceiverApplicationId(receiverId)
                .build()
    }

    override fun getAdditionalSessionProviders(context: Context): List<SessionProvider>? {
        return null
    }
}